utils/deploy.js

```diff
@@ -22,20 +22,30 @@ envalid.cleanEnv(process.env, {
     const dir = path.resolve(__dirname, '..', 'public');
     const branch = await garden.githubBranch();
     const production = branch === 'main';
-    const repository = await garden.githubRepository();
-    const commit = await garden.githubCommit();
-    const message = `https://github.com/${repository.owner}/${repository.repo}/commit/${commit}`;
-    const command = async () => {
-      const result = await garden.netlifyDeploy({
-        dir,
-        production,
-        message
-      });
-
-      return result;
-    };
-
-    const url = await garden.githubDeploy({ command, production });
+    const available = production ? Infinity : (await garden.netlifyBandwidth()).available;
+    const usage = production ? 0 : await garden.cmdDu(dir);
+    let url;
+
+    if (available > usage) {
+      const repository = await garden.githubRepository();
+      const commit = await garden.githubCommit();
+      const message = `https://github.com/${repository.owner}/${repository.repo}/commit/${commit}`;
+      const command = async () => {
+        const result = await garden.netlifyDeploy({
+          dir,
+          production,
+          message
+        });
+
+        return result;
+      };
+
+      url = await garden.githubDeploy({ command, production });
```

Comment:
Out of curiosity, if the GitHub deploy failed, would the error be bubbled up somewhere and handled?

Reply:
Yes. Note the caught, logged error below along with `process.exitCode = 1;` which would result in a CI fail.

Reply:
Thanks 👍🏽